src/std-traits/closures.md

```diff
@@ -10,39 +10,41 @@ implement special [`Fn`](https://doc.rust-lang.org/std/ops/trait.Fn.html),
 [`FnOnce`](https://doc.rust-lang.org/std/ops/trait.FnOnce.html) traits:
 
 ```rust,editable
-fn apply_with_log(func: impl FnOnce(i32) -> i32, input: i32) -> i32 {
-    println!("Calling function on {input}");
-    func(input)
+fn apply_and_log(func: impl FnOnce(i32) -> i32, func_name: &str, input: i32) {
+    println!("Calling {func_name}({input}): {}", func(input))
 }
 
 fn main() {
-    let add_3 = |x| x + 3;
-    println!("add_3: {}", apply_with_log(add_3, 10));
-    println!("add_3: {}", apply_with_log(add_3, 20));
+    let n = 3;
+    let add_3 = |x| x + n;
+    apply_and_log(&add_3, "add_3", 10);
+    apply_and_log(&add_3, "add_3", 20);
 
     let mut v = Vec::new();
     let mut accumulate = |x: i32| {
         v.push(x);
         v.iter().sum::<i32>()
     };
-    println!("accumulate: {}", apply_with_log(&mut accumulate, 4));
-    println!("accumulate: {}", apply_with_log(&mut accumulate, 5));
+    apply_and_log(&mut accumulate, "accumulate", 4);
+    apply_and_log(&mut accumulate, "accumulate", 5);
 
     let multiply_sum = |x| x * v.into_iter().sum::<i32>();
-    println!("multiply_sum: {}", apply_with_log(multiply_sum, 3));
+    apply_and_log(multiply_sum, "multiply_sum", 3);
 }
 ```
 
 <details>
 
-An `Fn` (e.g. `add_3`) neither consumes nor mutates captured values, or perhaps
-captures nothing at all. It can be called multiple times concurrently.
+An `Fn` (e.g. `add_3`) neither consumes nor mutates captured values. It can be
+called needing only a shared reference to the closure, which means the closure
+can be executed repeatedly and even concurrently.
 
 An `FnMut` (e.g. `accumulate`) might mutate captured values. You can call it
-multiple times, but not concurrently.
+multiple times, but not concurrently (because doing so would require having
```

Comment:
for parallelism, mention that it is called with an exclusive reference to the closure